codegen/codegen.go

```diff
@@ -31,8 +32,36 @@ func main() {
 		panic(err)
 	}
 
+	var jsonData map[string]interface{}
+	err = json.Unmarshal(b, &jsonData)
+	if err != nil {
+		panic(fmt.Errorf("failed to parse JSON: %v", err))
+	}
+	// rke doesn't need info about rke2 and k3s versions
+	delete(jsonData, "rke2")
+	delete(jsonData, "k3s")
+
+	b, err = json.Marshal(jsonData)
+	if err != nil {
+		panic(fmt.Errorf("failed to marshal json data: %v", err))
+	}
+
 	fmt.Println("Writing data")
 	if err := os.WriteFile(dataFile, b, 0755); err != nil {
 		return
 	}
 }
+
+func getURL() string {
+	u := os.Getenv(metadata.RancherMetadataURLEnv)
+	if u == "" {
+		u = defaultDevURL
+		tag := os.Getenv("TAG")
```

Comment:
The tag already exists so I'm hoping we don't need any other changes: https://github.com/rancher/rke/blob/2e0a4a74647262497ad72e30d7e9fadb499ab846/Dockerfile.dapper#L14

Reply:
actually we are not using dapper in the CI for rke. And i checked the action workflows, the TAG is not being set there so we need to make some changes in that i guess.
https://github.com/rancher/rke/blob/2e0a4a74647262497ad72e30d7e9fadb499ab846/.github/workflows/workflow.yaml#L40-L43
we did not use dapper in CI because there were some network related issues using dapper for integration tests (dind) is used in that.

Reply:
Thank you! Fixed to set `TAG` as env var in gha. 

I also noticed that `semver.NewVersion` errors out if tag starts with `v` so updated it and codegen_test.go to reflect this change. Could you re-review?